File: output/z_ai_glm_5/few/cpp_noimg/Chocolate em barra.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    if (!(cin >> n)) return 0;

    int x1, y1, x2, y2;
    cin >> x1 >> y1 >> x2 >> y2;

    int half = n / 2;

    bool possible = false;

    // Check horizontal cut (between rows 'half' and 'half + 1')
    // One sticker must be in the top half (row <= half) and the other in the bottom half (row > half)
    if ((x1 <= half && x2 > half) || (x2 <= half && x1 > half)) {
        possible = true;
    }

    // Check vertical cut (between columns 'half' and 'half + 1')
    // One sticker must be in the left half (col <= half) and the other in the right half (col > half)
    if ((y1 <= half && y2 > half) || (y2 <= half && y1 > half)) {
        possible = true;
    }

    if (possible) {
        cout << "S" << endl;
    } else {
        cout << "N" << endl;
    }

    return 0;
}

File: output/cbsoft_sbes_2026/anthropic_claude_sonnet_4_6/zero/python_img/Chocolate em barra.py
n = int(input())
x1, y1 = map(int, input().split())
x2, y2 = map(int, input().split())

half = n // 2

# Check horizontal cut: one figurinha in rows 1..half and other in rows half+1..n
horizontal = (x1 <= half and x2 > half) or (x2 <= half and x1 > half)

# Check vertical cut: one figurinha in cols 1..half and other in cols half+1..n
vertical = (y1 <= half and y2 > half) or (y2 <= half and y1 > half)

if horizontal or vertical:
    print("S")
else:
    print("N")